src/arrayMethodSort.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const n = this.length;
+
+    for (let i = 0; i < n - 1; i++) {
+      for (let j = 0; j < n - 1 - i; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[j], this[j + 1]) > 0) {
+            [this[j], this[j + 1]] = [this[j + 1], this[j]];
+          }
+        } else {
+          if (String(this[j]) > String(this[j + 1])) {
```

Comment:
When converting elements to strings, consider using `String.prototype.localeCompare` for the comparison, as it will handle non-ASCII characters more accurately.